File: graphify/tests/fixtures/sample.rb
require 'json'
require 'net/http'

class ApiClient
  def initialize(base_url)
    @base_url = base_url
  end

  def get(path)
    fetch(path, 'GET')
  end

  def post(path, body)
    fetch(path, 'POST')
  end

  private

  def fetch(path, method)
    uri = URI(@base_url + path)
    Net::HTTP.get(uri)
  end
end

def parse_response(raw)
  JSON.parse(raw)
end

File: graphify/tests/fixtures/sample.php
<?php

namespace App\Http;

use App\Auth\Authenticator;
use App\Cache\CacheManager;

class ApiClient
{
    private string $baseUrl;
    private Authenticator $auth;

    public function __construct(string $baseUrl)
    {
        $this->baseUrl = $baseUrl;
        $this->auth = new Authenticator();
    }

    public function get(string $path): string
    {
        return $this->fetch($path, 'GET');
    }

    public function post(string $path, string $body): string
    {
        return $this->fetch($path, 'POST');
    }

    private function fetch(string $path, string $method): string
    {
        $token = $this->auth->getToken();
        return $method . ' ' . $this->baseUrl . $path;
    }
}

function parseResponse(string $raw): array
{
    return json_decode($raw, true);
}
